src/NexusMods.DataModel/Settings/MnemonicDBSettingsBackend.cs

```diff
@@ -0,0 +1,105 @@
+using System.Text.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.Abstractions.Serialization.DataModel;
+using NexusMods.Abstractions.Settings;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.MnemonicDB.Abstractions.TxFunctions;
+
+namespace NexusMods.DataModel.Settings;
+
+/// <summary>
+/// A settings backend that uses the MnemonicDB for storage.
+/// </summary>
+public class MnemonicDBSettingsBackend : ISettingsStorageBackend
+{
+    private readonly ILogger<MnemonicDBSettingsBackend> _logger;
+    private readonly Lazy<IConnection> _conn;
+    private readonly Lazy<JsonSerializerOptions> _jsonOptions;
+    private readonly Lazy<IRepository<Setting.Model>> _settingRepository;
+
+    public MnemonicDBSettingsBackend(ILogger<MnemonicDBSettingsBackend> logger, IServiceProvider serviceProvider)
+    {
+        _logger = logger;
+
+        // NOTE(erri120): Lazy to fix circular dependency:
+        // SettingsManager -> DataStoreSettingsBackend -> IDataStore -> ISettingsManager
```

Comment:
```suggestion
```

Reply:
I think this code still needs to use Lazy, update the comment with something more fitting.